src/arrow1/reader.rs

```diff
@@ -1,32 +1,31 @@
 use crate::arrow1::error::Result;
 use arrow::ipc::writer::StreamWriter;
+use arrow::record_batch::RecordBatchReader;
 use bytes::Bytes;
-use parquet::arrow::{ArrowReader, ParquetFileArrowReader};
-use parquet::file::reader::{FileReader, SerializedFileReader};
-use std::sync::Arc;
+use parquet::arrow::arrow_reader::ParquetRecordBatchReaderBuilder;
 
 /// Internal function to read a buffer with Parquet data into a buffer with Arrow IPC Stream data
 /// using the arrow and parquet crates
 pub fn read_parquet(parquet_file: &[u8]) -> Result<Vec<u8>> {
     // Create Parquet reader
     let cursor = Bytes::copy_from_slice(parquet_file);
```

Comment:
unrelated: I realized today looking back at this that this is probably an accidental memory copy of the entire parquet buffer. Taking `Vec<u8>` as the param as the latest `main` does should hopefully improve memory usage here if I'm right